zer0-os/zOS
# feat: enhance event timeline processing to include create room event

### What does this do?
- This update modifies the getAllMessagesFromRoom method to ensure that room creation events are included when fetching room events. The method now explicitly checks for the existence of a room creation event and adds it to the timeline events.

### Why are we making this change?
- Previously, room creation events were sometimes omitted in the event timeline, leading to incomplete room histories when processing raw events to messages. By including room creation events, this change ensures that admin messages will render in the conversation item preview when fetching conversations. (Before and After below).

### How do I test this?
- with another user, create a room with your current user > when the conversation renders in the conversation list panel, you should see the admin message in the preview > when refreshing this should persist. 

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

BEFORE:

https://github.com/zer0-os/zOS/assets/39112648/e29065c6-d288-43f4-9ed3-d99b783c9e9f

AFTER:

https://github.com/zer0-os/zOS/assets/39112648/69ad691c-ee7a-48ee-8058-abc45309dcf3
